### PR TITLE
Deleted changeset handling in BriefcaseManager

### PR DESCRIPTION
**Pull request by austeja-bentley**
- Added handling in BriefcaseManager.pullAndMergeChanges to return a specific error when the latest local briefcase changeSet does not exist anymore
- Couple of checks if the cache file exists in ConcurrencyControl to be able to close and delete a briefcase which failed the most recent change push.
- Added an integration test to mimic the deleted changeSet workflow: user acquires a briefcase, pushes a few changes (one of which is invalid (does not apply when generating a checkpoint in iModelHub)), attempts to generate a checkpoint (by creating a named version), waits for its failure, deletes the invalid changeSet, waits for changeSet deletion completion, purges the local briefcase, re-downloads a new one and pushes a new changeSet.

Note: this is still in the works since we need a new specific test user added to the test project.

### CODE REVIEW

**Comment by pmconne** [comment]
Is this ready for review and merge?

**Comment by austeja-bentley** [comment]
> Is this ready for review and merge?

It is. I have pinged @ramanujam-raman to take a look at this since we have previously discussed the changes with him but I suppose the changes aren't too specific and could be reviewed by anyone.

**Comment by pmconne** [comment]
> I have pinged @ramanujam-raman to take a look at this since we have previously discussed the changes with him but I suppose the changes aren't too specific and could be reviewed by anyone.

Please ask @ramanujam-raman to review this so it can get merged.

**Comment by calebmshafer** [comment]
@austeja-bentley you waiting for a review on this?  If so, can you resolve the conflicts and I'll take a look tomorrow.

File: core/backend/src/test/integration/BriefcaseManager.test.ts [hunk 1]
```diff
@@ -661,4 +661,77 @@ describe("BriefcaseManager (#integration)", () => {
     requestContext.enter();
   });
 
+  it("Should be able to recover after changeSet deletion (#integration)", async () => {
+    const testUser = {
+      email: Config.App.getString("imjs_test_imodelhub_user_name"),
```
[the remark this answers is not in the document]
Comment (austeja-bentley) [reply core/backend/src/test/integration/BriefcaseManager.test.ts:668]: Added

**Review by calebmshafer — APPROVED** [review]
This looks good to me. I am a bit worried about how long the test could potentially take (max is 10 minutes).  Let's merge this in and see how it goes for a few days. If it starts taking too long we can split into a separate long-running integration tests job like we discussed previously.

**Pull request merged**